tests/c_languages/CPPCheckBearTest.py

```diff
@@ -18,8 +18,8 @@
 
 bad_file = """
 #define f(c) { \
-    char *p = new char[10];  \
-    p[c] = 42; \
+    char s[10]; \
```

Comment:
How does that help?
I'm not sure it does.  I've done a bit of digging around in their code, to determine exactly where "BufferOverrun" and "MemoryLeak" are skipping the processing of this C++ code when the file extension isn't provided (or is unknown). In both cases the condition is `Path::isCPP()`.
As result, as far as I can see, a test using `.cpp` extension with some bad C++ code, will only verify that:

1.  using a `.cpp` extension enables `cppcheck` C++ language features that are inside `Path::isCPP()` conditional blocks, which is a very simple function of `cppcheck`, and it has its own unit tests covering that.  Also it has not to our knowledge ever failed.
2.  the `cppcheck` C++ feature checker used is functioning correctly.  Again, the `cppcheck` unit tests should be covering this.  And if it does fail; then what?

Also worth noting that since the current `bad_file` is triggering errors by both "BufferOverrun" and "MemoryLeak" checkers, but only looking for the existence of one or more errors, it currently does not verify the correctness of either, as either one can break and our test will still pass.  We need a way to tell `verify_local_bear` how many expected errors there are for each file in `invalid_files`, or ... my preference .. create a dedicated test helper that verifies the list of errors matches the expected errors, with fuzzy matching for cases where the checker program improves its emitted messages in latter releases.